networkx/drawing/layout.py

```diff
@@ -1490,16 +1493,17 @@ def estimate_factor(n, swing, traction, speed, speed_efficiency, jitter_toleranc
 
         # update pos
         if adjust_sizes:
-            swinging = mass * np.linalg.norm(update, axis=-1)
-            factor = 0.1 * speed / (1 + np.sqrt(speed * swinging))
             df = np.linalg.norm(update, axis=-1)
+            swinging = mass * df
+            factor = 0.1 * speed / (1 + np.sqrt(speed * swinging))
```

Comment:
move the computation of `df` to before computing swinging to avoid recomputing `norm(update, axis=-1)`